### PR TITLE
add type components

### PR DESCRIPTION
**Pull request by afebbraro**
## What does this PR do?
Adds two new components(directives in angular) to both the spark-react and spark-angular packages for SprkHeading(sprk-heading) and for SprkText(sprk-text) for a total of four new components.

### Associated Issue
Fixes #2769 

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR
then please remove it.

### Documentation
 - [x] Update Spark Docs

### Code
 - [x] Build Component in Angular
 - [x] Build Component in React
 - [x] Unit Testing in Angular with `npm run test` in `angular/` (100% coverage, 100% passing)
 - [x] Unit Testing in React with `npm run test` in `react/` (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [ ] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [ ] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [ ] Apple Safari (Mobile)

### Manual Testing Plan
  - [ ] Update excel sheet with component changes

### Screenshots
Add screenshots to help explain your PR if you'd like. However, this is not
expected.
Screen Shot 2020-02-13 at 9 16 29 AM
Screen Shot 2020-02-13 at 9 16 22 AM

### CODE REVIEW

File: angular/projects/spark-angular/src/lib/directives/sprk-heading/sprk-heading.directive.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  @HostBinding('class.sprk-b-PageTitle')
+  /**
+   * Value will be used to select
+   * the heading style.
+   */
+  @Input() isPageTitle = false;
+}
+
+
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/directives/sprk-heading/sprk-heading.directive.ts]: Extra line here at the end.

File: angular/projects/spark-angular/src/lib/directives/sprk-text/sprk-text.stories.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      )
+    )
+  ],
+  parameters: {
+    info: `
+${markdownDocumentationLinkBuilder('typography', 'foundations')}
+
+##### Line-height Collision
+
+By default, line height on type will create extra
```
Comment (RCopeland) [inline angular/projects/spark-angular/src/lib/directives/sprk-text/sprk-text.stories.ts]: line-height

File: react/src/base/typography/SprkHeading/SprkHeading.stories.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    SprkText,
+  },
+  parameters: {
+    jest: ['SprkHeading'],
+    info: `
+${markdownDocumentationLinkBuilder('typography', 'foundations')}
+
+##### Line-height Collision
+
+By default, line height on type will create extra
```
Comment (RCopeland) [inline react/src/base/typography/SprkHeading/SprkHeading.stories.js]: line-height

Comment (afebbraro) [reply react/src/base/typography/SprkHeading/SprkHeading.stories.js]: @RCopeland , we have this on the gatsby site like I have it in this PR. https://sparkdesignsystem.com/using-spark/foundations/typography/#gatsby-focus-wrapper - I'm going to keep it like we have it in the gatsby site.

File: angular/projects/spark-angular/src/lib/directives/sprk-text/sprk-text.stories.ts (hunk 2 above)
Comment (afebbraro) [reply angular/projects/spark-angular/src/lib/directives/sprk-text/sprk-text.stories.ts]: @RCopeland , we have this on the gatsby site like I have it in this PR. https://sparkdesignsystem.com/using-spark/foundations/typography/#gatsby-focus-wrapper - I'm going to keep it like we have it in the gatsby site.

**Pull request merged**